### PR TITLE
Support offline contributions, configurable sequencer URL & direct hex-encoded entropy flag

### PR DESCRIPTION
**Pull request by jsign**
This PR includes:
- New CLI commands for offline contributions [see tentative rendered README](https://github.com/jsign/go-kzg-ceremony-client/blob/278affdc204907c48783e452c4e7fdf4e9a31907/README.md#offline-contributions).
- Mentioned projects (that I know of) that have used this client, [see rendered](https://github.com/jsign/go-kzg-ceremony-client/blob/278affdc204907c48783e452c4e7fdf4e9a31907/README.md#ethereum-eip-4844-powers-of-tau-ceremony-client).
- New `--sequencer-url`, which allows targeting testing environments.
- New `--hex-entropy` allows passing entropy as hex-encoded bytes, which can be more convenient in air-gapped devices than the "pull URL" option (which still exists).

Multiple _Special Contributions_ projects (at this point three) are testing and providing feedback on this branch.
If you're reading this and interested in offline contributions, [ping me via DM](https://twitter.com/ignaciohagopian).

### CODE REVIEW

**Comment by jflo** [comment]
I've setup a local sequencer for development on my network. I can use this branch just fine to make contributions in online mode. Offline mode is failing with a 400 though, using the same session-id that works in online mode. I have configured the sequencer to accept multiple contributions from the same user, and I can submit multiple with the same session id in online mode.

I get the current state using:

`kzgcli offline download-state localnet-state.json --sequencer-url <localnet-sequencer-url>`

contribution on top of that state seems fine...

`kzgcli offline contribute localnet-state.json contrib.json --hex-entropy "0xDEADBEEF"`

but get 400s on

`kzgcli offline send-contribution contrib.json --session-id <session-id> --sequencer-url <localnet-sequencer-url>`

So feels like my sequencer is setup right, but perhaps the offline send-contrib might not be passing in the session-id correctly?

**Comment by jsign** [comment]
> So feels like my sequencer is setup right, but perhaps the offline send-contrib might not be passing in the session-id correctly?

The thing is that in "online mode" the flow is:
- Clients ask for their turns.
- The sequencer randomly decides is _your_ turn.
- _Now that you're allowed_ to send the contribution, you can send it.

In this client in "online mode", we do that flow. But in "offline mode" the `kzgcli offline send-contribution` **assumes** is your turn, which probably your sequencer isn't expecting.

If I add to the offline mode to also "ask for its turn" probably it will work, but I'm not totally sure that's the way that "Sepecial contributions" will work. Since "Special contributions" are known and in many cases they're in very constraint environments, I'd expect the sequencer to already assign a defined "slot" where users can send their contribution directly (because their turn was defined).

So TL;DR, depending on how "taking turns" work in "Special contributions" I might have to tweak the client:
- If still users have to "ask for turns", then I'll have to add that (quite easy change since it's exactly what online mode does)
- If that isn't needed because you're assigned a defined time-span, then the current version is fine and you should send it when appropriate.

So now you local sequencer is working in "online mode" and is expecting to "ask for turns".

If you know something else about how this will work for "Special contributions", let me know and we can verify or adjust!

**Comment by jsign** [comment]
@jflo, I added the code that asks for your turn. From what Carl said, it should always succeed if you're in your reserved slot period.

Do you mind trying again in your sequencer?

**Comment by jsign** [comment]
If all goes well, I'll merge this into `main`.

**Comment by jsign** [comment]
Considering the online phase has ~finished, actually the `main` branch only makes sense for offline contributions, so I'll mege!
We can fix stuff if needed!
